### ISSUE TITLE
Perl Warnings on Custom Annotation from gnomAD exomes

### ISSUE THREAD

**Issue by james-lawlor** [Ensembl/ensembl-vep#97]
Hi,
When using VEP v. 90.5 to add custom annotations from the ensembl-provided gnomAD exome VCF, I'm running into a large number of perl warnings. I've got VEP set up to use nearly all the annotations listed in the gnomAD exome VCF header.

There do not seem to be any other negative impacts on the output, but with the number of warnings I thought it might be worth mentioning. I haven't yet been able to narrow down which annotation(s) may be responsible. (I haven't had the same problem from the genome VCF).

Here's there first 5000 warnings I get from annotating a typical VCF:

[warnings_5000.txt](https://github.com/Ensembl/ensembl-vep/files/1451249/warnings_5000.txt)

Toy example:
```
[jlawlor@login01 test_vep]$ vep -i tiny.vcf -o tiny_vep.vcf --config test_perl_errors.ini 
Use of uninitialized value $value in pattern match (m//) at /gpfs/gpfs2/software/vep-90.5/modules/Bio/EnsEMBL/VEP/AnnotationSource/File/VCF.pm line 181, <__ANONIO__> line 54628.
Use of uninitialized value in join or string at /gpfs/gpfs2/software/vep-90.5/modules/Bio/EnsEMBL/VEP/Utils.pm line 183, <__ANONIO__> line 54628.
```
tiny.vcf:
```
1       13656   .       CAG     C       54267.10        VQSRTrancheINDEL99.00to99.90
```

test_perl_errors.ini:
```
custom /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz,gnomadEx,vcf,exact,0,AC,AF,AN,BaseQRankSum,ClippingRankSum,DB,DP,FS,InbreedingCoeff,MQ,MQRankSum,QD,ReadPosRankSum,SOR,VQSLOD,VQSR_culprit,VQSR_NEGATIVE_TRAIN_SITE,VQSR_POSITIVE_TRAIN_SITE,AC_AFR,AC_AMR,AC_ASJ,AC_EAS,AC_FIN,AC_NFE,AC_OTH,AC_SAS,AC_Male,AC_Female,AN_AFR,AN_AMR,AN_ASJ,AN_EAS,AN_FIN,AN_NFE,AN_OTH,AN_SAS,AN_Male,AN_Female,AF_AFR,AF_AMR,AF_ASJ,AF_EAS,AF_FIN,AF_NFE,AF_OTH,AF_SAS,AF_Male,AF_Female,GC_AFR,GC_AMR,GC_ASJ,GC_EAS,GC_FIN,GC_NFE,GC_OTH,GC_SAS,GC_Male,GC_Female,AC_raw,AN_raw,AF_raw,GC_raw,GC,Hom_AFR,Hom_AMR,Hom_ASJ,Hom_EAS,Hom_FIN,Hom_NFE,Hom_OTH,Hom_SAS,Hom_Male,Hom_Female,Hom_raw,Hom,STAR_AC,STAR_AC_raw,STAR_Hom,POPMAX,AC_POPMAX,AN_POPMAX,AF_POPMAX,DP_MEDIAN,DREF_MEDIAN,GQ_MEDIAN,AB_MEDIAN,AS_RF,AS_RF_POSITIVE_TRAIN,AS_RF_NEGATIVE_TRAIN,AC_AFR_Male,AC_AMR_Male,AC_ASJ_Male,AC_EAS_Male,AC_FIN_Male,AC_NFE_Male,AC_OTH_Male,AC_SAS_Male,AC_AFR_Female,AC_AMR_Female,AC_ASJ_Female,AC_EAS_Female,AC_FIN_Female,AC_NFE_Female,AC_OTH_Female,AC_SAS_Female,AN_AFR_Male,AN_AMR_Male,AN_ASJ_Male,AN_EAS_Male,AN_FIN_Male,AN_NFE_Male,AN_OTH_Male,AN_SAS_Male,AN_AFR_Female,AN_AMR_Female,AN_ASJ_Female,AN_EAS_Female,AN_FIN_Female,AN_NFE_Female,AN_OTH_Female,AN_SAS_Female,AF_AFR_Male,AF_AMR_Male,AF_ASJ_Male,AF_EAS_Male,AF_FIN_Male,AF_NFE_Male,AF_OTH_Male,AF_SAS_Male,AF_AFR_Female,AF_AMR_Female,AF_ASJ_Female,AF_EAS_Female,AF_FIN_Female,AF_NFE_Female,AF_OTH_Female,AF_SAS_Female,GC_AFR_Male,GC_AMR_Male,GC_ASJ_Male,GC_EAS_Male,GC_FIN_Male,GC_NFE_Male,GC_OTH_Male,GC_SAS_Male,GC_AFR_Female,GC_AMR_Female,GC_ASJ_Female,GC_EAS_Female,GC_FIN_Female,GC_NFE_Female,GC_OTH_Female,GC_SAS_Female,Hemi_AFR

format vcf
vcf TRUE
pick_allele TRUE
cache TRUE
offline TRUE
merged TRUE
everything TRUE
dir_cache /gpfs/gpfs2/reference/vep/
cache_version 90
assembly GRCh37
use_given_ref TRUE
fasta /gpfs/gpfs1/myerslab/reference/genomes/bwa-0.7.8/GRCh37.fa
force_overwrite TRUE
check_existing TRUE
```

gnomAD exome annotation at that location:
```
tabix /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz 1:13656-13656
1	13656	.	CAG	C	422117.14	RF	AC=670;AF=2.79866e-02;AN=23940;BaseQRankSum=-2.77100e+00;ClippingRankSum=-1.24000e-01;DP=1138707;FS=0.00000e+00;InbreedingCoeff=-6.07000e-02;MQ=2.28200e+01;MQRankSum=1.42000e+00;QD=6.08000e+00;ReadPosRankSum=-2.79900e+00;SOR=5.43000e-01;VQSLOD=-1.43200e+00;VQSR_culprit=FS;VQSR_NEGATIVE_TRAIN_SITE;GQ_HIST_ALT=10|109|71|143|79|54|70|42|30|21|40|36|54|68|63|64|72|53|49|939;DP_HIST_ALT=208|333|121|226|329|261|175|127|95|72|40|34|17|17|7|2|1|1|1|0;AB_HIST_ALT=0|49|436|384|276|199|86|43|25|8|8|2|12|22|5|28|47|31|7|0;GQ_HIST_ALL=15465|4730|1254|1162|774|653|688|649|613|672|601|453|761|410|630|367|650|291|523|4501;DP_HIST_ALL=13566|1460|1400|2651|3140|2639|2438|2145|1879|1323|1032|702|486|315|216|154|94|69|45|32;AB_HIST_ALL=0|49|436|384|276|199|86|43|25|8|8|2|12|22|5|28|47|31|7|0;AC_AFR=24;AC_AMR=159;AC_ASJ=37;AC_EAS=19;AC_FIN=16;AC_NFE=308;AC_OTH=21;AC_SAS=86;AC_Male=357;AC_Female=313;AN_AFR=3748;AN_AMR=3652;AN_ASJ=390;AN_EAS=5854;AN_FIN=244;AN_NFE=6958;AN_OTH=604;AN_SAS=2490;AN_Male=12402;AN_Female=11538;AF_AFR=6.40342e-03;AF_AMR=4.35378e-02;AF_ASJ=9.48718e-02;AF_EAS=3.24564e-03;AF_FIN=6.55738e-02;AF_NFE=4.42656e-02;AF_OTH=3.47682e-02;AF_SAS=3.45382e-02;AF_Male=2.87857e-02;AF_Female=2.71278e-02;GC_AFR=1850,24,0;GC_AMR=1670,153,3;GC_ASJ=159,35,1;GC_EAS=2908,19,0;GC_FIN=107,14,1;GC_NFE=3183,284,12;GC_OTH=282,19,1;GC_SAS=1161,82,2;GC_Male=5856,333,12;GC_Female=5464,297,8;AC_raw=2466;AN_raw=71694;AF_raw=3.43962e-02;GC_raw=33780,1668,399;GC=11320,630,20;Hom_AFR=0;Hom_AMR=3;Hom_ASJ=1;Hom_EAS=0;Hom_FIN=1;Hom_NFE=12;Hom_OTH=1;Hom_SAS=2;Hom_Male=12;Hom_Female=8;Hom_raw=399;Hom=20;POPMAX=ASJ;AC_POPMAX=37;AN_POPMAX=390;AF_POPMAX=9.48718e-02;DP_MEDIAN=22;DREF_MEDIAN=1.58426e-13;GQ_MEDIAN=86;AB_MEDIAN=1.90476e-01;AS_RF=1.12833e-01;AS_FilterStatus=RF
```

Let me know if I can provide any more examples or files that would be helpful. 

Note: ensembl-provided gnomAD vcfs downloaded from:
[https://www.ensembl.org/info/docs/tools/vep/script/vep_example.html#gnomad]

**Comment by willmclaren** [comment]
The warnings are appearing because you're requesting the field "VQSR_NEGATIVE_TRAIN_SITE"; this does not have a value in the VCF (as key=value e.g. AC=670), it merely appears as a key on its own.

In designing this feature we hadn't anticipated that users might want to include such fields. How would you want this to appear in the resulting annotation, or is it being included mistakenly?

One suggestion would be that we substitute in a basic true value of `1` for such keys.

**Comment by james-lawlor** [comment]
Substituting a basic true value like 1 or TRUE sounds perfectly reasonable to me.

**Comment by james-lawlor** [comment]
Substituting a basic true value like 1 or TRUE sounds perfectly reasonable to me.

**Comment by willmclaren** [comment]
Fixed now in release/90.8, VEP will substitute in a value of `1` for INFO fields with no value.
